blog/2018-07-17-UnitTestsInXslxFileForConferenceScheduling.adoc

```diff
@@ -0,0 +1,122 @@
+            .withPreferredRoomTagSet(Collections.emptySet())
+            .withProhibitedRoomTagSet(Collections.emptySet())
+            .withUndesiredRoomTagSet(Collections.emptySet())
+            .withRequiredTimeslotTagSet(Collections.emptySet())
+            .withPreferredTimeslotTagSet(Collections.emptySet())
+            .withProhibitedTimeslotTagSet(Collections.emptySet())
+            .withUndesiredTimeslotTagSet(Collections.emptySet());
+    Talk talk2 = new Talk(2L)
+            .withTalkType(talkType)
+            .withSpeakerList(Collections.emptyList())
+            .withRequiredRoomTagSet(Collections.emptySet())
+            .withPreferredRoomTagSet(Collections.emptySet())
+            .withProhibitedRoomTagSet(Collections.emptySet())
+            .withUndesiredRoomTagSet(Collections.emptySet())
+            .withRequiredTimeslotTagSet(Collections.emptySet())
+            .withPreferredTimeslotTagSet(Collections.emptySet())
+            .withProhibitedTimeslotTagSet(Collections.emptySet())
+            .withUndesiredTimeslotTagSet(Collections.emptySet());
+    LocalDateTime start1 = LocalDateTime.of(2018, 1, 1, 9, 0);
+    LocalDateTime end1 = LocalDateTime.of(2018, 1, 1, 10, 0);
+    LocalDateTime start2 = LocalDateTime.of(2018, 1, 1, 9, 30);
+    LocalDateTime end2 = LocalDateTime.of(2018, 1, 1, 10, 30);
+    LocalDateTime start3 = LocalDateTime.of(2018, 1, 1, 10, 0);
+    LocalDateTime end3 = LocalDateTime.of(2018, 1, 1, 11, 0);
+    Timeslot slot1 = new Timeslot(1L)
+            .withTalkTypeSet(Collections.singleton(talkType))
+            .withStartDateTime(start1)
+            .withEndDateTime(end1);
+    Timeslot slot2 = new Timeslot(2L)
+            .withTalkTypeSet(Collections.singleton(talkType))
+            .withStartDateTime(start2)
+            .withEndDateTime(end2);
+    Timeslot slot3 = new Timeslot(3L)
+            .withTalkTypeSet(Collections.singleton(talkType))
+            .withStartDateTime(start3)
+            .withEndDateTime(end3);
+    Room room1 = new Room(1L)
+            .withTalkTypeSet(Collections.singleton(talkType))
+            .withUnavailableTimeslotSet(Collections.emptySet());
+    ConferenceSolution solution = new ConferenceSolution(1L)
+            .withTalkTypeList(Collections.singletonList(talkType))
+            .withTalkList(Arrays.asList(talk1, talk2))
+            .withTimeslotList(Arrays.asList(slot1, slot2, slot3))
+            .withRoomList(Collections.singletonList(room1))
+            .withSpeakerList(Collections.emptyList());
+    scoreVerifier.assertHardWeight(ROOM_CONFLICT, 0, solution);
+    // talks in same room without overlapping time slots
+    talk1.withRoom(room1).withTimeslot(slot1);
+    talk2.withRoom(room1).withTimeslot(slot3);
+    scoreVerifier.assertHardWeight(ROOM_CONFLICT, 0, solution);
+    // talks in same room with overlapping time slots
+    talk2.withTimeslot(slot2);
+    scoreVerifier.assertHardWeight(ROOM_CONFLICT, -10, solution);
+    }
```

Comment:
This test is long. It doesn't need to be complete, it just needs to be complete enough to serve as a "BEFORE" example, to compare with the "AFTER" case.
Can we shorten it or replace parts with "..." to make it less verbose?